controller/src/beerocks/bml/internal/bml_internal.cpp

```diff
@@ -1206,6 +1251,309 @@ int bml_internal::bml_get_vap_list_credentials(BML_VAP_INFO *vaps, uint8_t &vaps
+    }
+
+    // Initialize the promise for receiving the response
+    beerocks::promise<bool> prmChannelScanParamsGet;
+    m_prmChannelScanParamsGet = &prmChannelScanParamsGet;
+    int iOpTimeout            = RESPONSE_TIMEOUT; // Default timeout
+
+    beerocks_message::sChannelScanRequestParams ScanParams;
+    m_scan_params = &ScanParams;
+
+    //CMDU message
+    auto request = message_com::create_vs_message<
+        beerocks_message::cACTION_BML_CHANNEL_SCAN_GET_CONTINUOUS_PARAMS_REQUEST>(cmdu_tx);
+
+    if (!request) {
+        LOG(ERROR)
+            << "Failed building ACTION_BML_CHANNEL_SCAN_GET_CONTINUOUS_PARAMS_REQUEST message!";
+        return (-BML_RET_OP_FAILED);
+    }
+
+    request->radio_mac() = network_utils::mac_from_string(mac);
+    // Build and send the message
+    if (!message_com::send_cmdu(m_sockMaster, cmdu_tx)) {
+        LOG(ERROR) << "Failed sending param get message!";
+        m_prmChannelScanParamsGet = nullptr;
+        m_scan_params             = nullptr;
+        return (-BML_RET_OP_FAILED);
+    }
+    LOG(DEBUG) << "ACTION_BML_CHANNEL_SCAN_GET_CONTINUOUS_PARAMS_REQUEST sent";
+
+    int iRet = BML_RET_OK;
+
+    if (!m_prmChannelScanParamsGet->wait_for(iOpTimeout)) {
+        LOG(WARNING) << "Timeout while waiting for param get response...";
+        iRet = -BML_RET_TIMEOUT;
+    }
+
+    // Clear the scan params member
+    m_scan_params = nullptr;
+
+    // Clear the promise holder
+    m_prmChannelScanParamsGet = nullptr;
+
+    if (output_dwell_time != nullptr) {
+        *output_dwell_time = ScanParams.dwell_time_ms;
+    }
+    if (output_interval_time != nullptr) {
+        *output_interval_time = ScanParams.interval_time_sec;
+    }
+    if (output_channel_pool_size != nullptr && output_channel_pool != nullptr) {
+        *output_channel_pool_size = ScanParams.channel_pool_size;
+        if (*output_channel_pool_size > BML_CHANNEL_SCAN_MAX_CHANNEL_POOL_SIZE) {
+            LOG(WARNING) << "Channel pool size is too big for the allocated channel pool...";
+            *output_channel_pool_size = BML_CHANNEL_SCAN_MAX_CHANNEL_POOL_SIZE;
+            /* Currently this simply cuts the given channel pool to the maximal size
+               If this is considered an error uncomment the following line */
+            // iRet = -BML_RET_OP_FAILED;
+        }
+        std::copy_n(ScanParams.channel_pool, *output_channel_pool_size, output_channel_pool);
+    }
+
+    if (iRet != BML_RET_OK) {
+        LOG(ERROR) << "Params get failed!";
+        return (iRet);
+    }
+
+    return (iRet);
+}
+
+int bml_internal::get_dcs_scan_results(const std::string &mac, BML_NEIGHBOR_AP **output_results,
+                                       unsigned int *output_results_size,
+                                       const unsigned int max_results_size,
+                                       uint8_t *output_result_status, bool is_single_scan)
+{
+    if (m_sockPlatform == nullptr && !connect_to_platform()) {
```

Comment:
compare pointers with (! pointer), not `==nullptr`.

Reply:
Fixed